test/test_echo_server.rs

```diff
@@ -168,20 +157,19 @@ impl EchoClient {
                 debug!("CLIENT : We read {} bytes!", r);
 
                 // prepare for reading
-                let mut buf = buf.flip();
+                for actual in buf.iter() {
+                    let expect = self.rx[0];
+                    self.rx = &self.rx[1..];
 
-                while buf.has_remaining() {
-                    let actual = buf.read_byte().unwrap();
-                    let expect = self.rx.read_byte().unwrap();
-
-                    assert!(actual == expect, "actual={}; expect={}", actual, expect);
+                    assert!(*actual == expect, "actual={}; expect={}", actual, expect);
                 }
 
-                self.mut_buf = Some(buf.flip());
+                buf.clear();
```

Comment:
Same as above, I think this needs to be `buf.drain(..r)`.

Reply:
This should be okay, as below, highest layer with same for-loop above.